sound/soc/sof/control.c

```diff
@@ -266,7 +266,7 @@ int snd_sof_enum_put(struct snd_kcontrol *kcontrol,
 	if (pm_runtime_active(scomp->dev))
 		snd_sof_ipc_set_get_comp_data(scontrol,
 					      SOF_IPC_COMP_SET_VALUE,
-					      SOF_CTRL_TYPE_VALUE_CHAN_GET,
+					      SOF_CTRL_TYPE_VALUE_CHAN_SET,
```

Comment:
there should be a Fixes tag @RanderWang, so that this is applied to linux stable.

Can you and @bardliao double check if the tag should be this

````
Fixes: 0c888baba8e0 ('ASoC: SOF: dont wake dsp up in kcontrol IO')
````

Reply:
> there should be a Fixes tag @RanderWang, so that this is applied to linux stable.
> 
> Can you and @bardliao double check if the tag should be this
> 
> ```
> Fixes: 0c888baba8e0 ('ASoC: SOF: dont wake dsp up in kcontrol IO')
> ```

`SOF_CTRL_TYPE_VALUE_CHAN_SET` was firstly used by
c3078f539704 ('ASoC: SOF: Add Sound Open Firmware KControl support')
What 0c888baba8e0 did is just copy and paste.

Reply:
updated PR, thanks

Reply:
@bardliao the question is not "what was the first commit", but what is the earliest commit this patch will apply on for backports. This patch will not apply on c3078f5 ('ASoC: SOF: Add Sound Open Firmware KControl support')

Reply:
@plbossart I'd argue that the fixes tag is not important here. Yes, its a typo but it doesnt realy "fix" anything. We got lucky in the sense that the get/set commands are handled with the same switch/case and the actual IPC command is a different one SOF_IPC_COMP_SET_VALUE/GET_VALUE which is actually set correctly for the put cae.

Reply:
Not able to follow @ranj063, usually when directions are inverted things can go bad. How is it not a fix?

Reply:
@plbossart the ctrl->type for GET/SET are handled by the same case as in:
https://github.com/thesofproject/linux/blob/85bfa34a88d6d31c5bbec50d2077f1fbc117b290/sound/soc/sof/ipc.c#L800
and
https://github.com/thesofproject/linux/blob/85bfa34a88d6d31c5bbec50d2077f1fbc117b290/sound/soc/sof/control.c#L665

And the actual IPC command is set correctly for put as in:
https://github.com/thesofproject/linux/blob/85bfa34a88d6d31c5bbec50d2077f1fbc117b290/sound/soc/sof/control.c#L146
and 
https://github.com/thesofproject/linux/blob/85bfa34a88d6d31c5bbec50d2077f1fbc117b290/sound/soc/sof/control.c#L219

So, the code works as expected even though there's a typo for the ctrl->cmd_type. It needs to be fixed but it hasnt broken anything so far.

Reply:
ok, so this is not a fix, and the commit subject and message need to explain that there is no functional change...

Reply:
Suggested edit:

"
SOF_CTRL_TYPE_VALUE_CHAN_SET should be used for put operations for consistency.

The current use of _GET is obviously incorrect but _GET and _SET result in the same action so there is no functional change introduced by this patch
"

Reply:
thanks very much!  updated.